### PR TITLE
feat: add glob pattern matching for env files

### PR DESCRIPTION
**Pull request by sapenlei**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

The envFilePath only supports matching specific paths, which becomes cumbersome when trying to match multiple paths.

Issue Number: N/A

## What is the new behavior?
Use glob pattern matching for paths. For example:
`envFilePath: ".env*"` can match .env, .env.local, .env.production, etc.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
   "dependencies": {
     "dotenv": "16.4.5",
     "dotenv-expand": "10.0.0",
+    "glob": "^11.0.0",
```
Comment (Papooch) [inline package.json:23]: You added a dependency to `package.json`, but did not commit the updated `package-lock.json`, so the tests are failing on CI. Please add the lock file.

File: lib/config.module.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ import { ConfigFactoryKeyHost } from './utils';
 import { createConfigProvider } from './utils/create-config-factory.util';
 import { getRegistrationToken } from './utils/get-registration-token.util';
 import { mergeConfigObject } from './utils/merge-configs.util';
+import { globSync } from 'glob';
```
Comment (Papooch) [inline lib/config.module.ts:22]: Imports should be sorted by scoped->unscoped->relative, and alphabetically. (in VSCode, there's a command called "Organize Imports" - Shift+Alt+O)

**Review by Papooch — COMMENTED** [review]
You might want to update [the docstring in the options interface](https://github.com/nestjs/config/blob/b2bb9b2d05b16435f40d47fc64cbd7fde558d1a1/lib/interfaces/config-module-options.interface.ts#L32) with a mention of supporting globs, too.

**Comment by kamilmysliwiec** [comment]
You can just run `glob` in your app and pass the output string array to the `ConfigModule` options - no need for adding this dependency to the package

**Pull request closed without merging**